restorer.go

```diff
@@ -4,33 +4,43 @@ import (
 	"github.com/pkg/errors"
 	"golang.org/x/sync/errgroup"
 
+	"github.com/buildpacks/imgutil"
+
 	"github.com/buildpacks/lifecycle/buildpack"
+
+	"github.com/buildpacks/lifecycle/api"
 	"github.com/buildpacks/lifecycle/layers"
 	"github.com/buildpacks/lifecycle/platform"
 )
 
 type Restorer struct {
-	LayersDir  string
-	Buildpacks []buildpack.GroupBuildpack
-	Logger     Logger
+	LayersDir     string
+	Buildpacks    []buildpack.GroupBuildpack
+	Logger        Logger
+	SkipLayers    bool
+	LayerAnalyzer LayerAnalyzer
+	PlatformAPI   *api.Version
 }
 
 // Restore attempts to restore layer data for cache=true layers, removing the layer when unsuccessful.
 // If a usable cache is not provided, Restore will remove all cache=true layer metadata.
-func (r *Restorer) Restore(cache Cache) error {
+func (r *Restorer) Restore(img imgutil.Image, cache Cache) error {
 	// Create empty cache metadata in case a usable cache is not provided.
-	var meta platform.CacheMetadata
-	if cache != nil {
-		var err error
-		if !cache.Exists() {
-			r.Logger.Info("Layer cache not found")
+	meta, err := r.LayerAnalyzer.RetrieveMetadataFrom(cache)
```

Comment:
This feels a bit weird to me in that if `!r.analyzesLayers()` I'd expect the `LayerAnalyzer` not to be used... could it be broken out as a separate responsibility?